compiler/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -141,14 +141,18 @@ impl<'a> ModCollector<'a> {
     fn collect_impls(&mut self, context: &mut Context, impls: Vec<TypeImpl>, krate: CrateId) {
         let module_id = ModuleId { krate, local_id: self.module_id };
 
-        for r#impl in impls {
+        for mut r#impl in impls {
             let mut unresolved_functions = UnresolvedFunctions {
                 file_id: self.file_id,
                 functions: Vec::new(),
                 trait_id: None,
                 self_type: None,
             };
 
+            for (method, _) in &mut r#impl.methods {
+                method.def.where_clause.append(&mut r#impl.where_clause);
```

Comment:
```suggestion
                method.def.where_clause.extend(r#impl.where_clause.clone());
```
`append` drains the Vec on the rhs, so this would only add the where clause to the first method in the impl

Reply:
ah yes good catch